src/interceptors/ClientRequest/NodeClientRequest.ts

```diff
@@ -229,6 +227,7 @@ export class NodeClientRequest extends ClientRequest {
           'encountered resolver exception, aborting request...',
           resolverResult.error
         )
+        this.responseSource = 'mock'
```

Comment:
This is incorrect. At this point, we only know that the listeners have finished running but there's no certainty whether they have returned anything. We should remove this particular line.

Reply:
IIUC, if we reach this line, the handler throws an error, and we want to send it back to the user.
I couldn't put it outside the `if` because of the reason you mentioned, so I put it twice inside each `if`.

Reply:
> If the response source remains unknown, and there's an error in the request listener, we will never forward it to the user, is that correct?

Yes, one of the tests failed because of this.

> Perhaps for error suppression we should have this.suppressErrors boolean flag?

Like the  `afterConnectionPhase` solution?

Reply:
It looks like we can reuse the same `this.requestSent` new internal flag I talk about in [here](https://github.com/mswjs/interceptors/pull/462#discussion_r1343809809) for this purpose too! 

I believe once the request is sent (`.end()` is called), any other errors that arise should never be suppressed. Those errors are either:

1. Errors originating from the user-defined request listeners. 
2. Errors originating from Node when performing a bypassed request.